src/attached_probe.cpp

```diff
@@ -811,48 +811,56 @@ void AttachedProbe::load_prog(BPFfeature &feature)
         if (bt_debug == DebugLevel::kNone)
           silencer.silence();
 
-        LIBBPF_OPTS(bpf_btf_load_opts,
-                    btf_opts,
-                    .log_buf = log_buf.get(),
-                    .log_level = static_cast<__u32>(log_level),
-                    .log_size = static_cast<__u32>(log_buf_size), );
-
-        auto &btf = prog_.getBTF();
-        auto btf_fd = bpf_btf_load(btf.data(), btf.size(), &btf_opts);
-
-        opts.prog_btf_fd = btf_fd;
+        // Not all older kernels cope well with BTF loaded (see #2913):
+        // On the 3rd attempt, retry bpf_prog_load() without loading BTF
+        int btf_fd = -1;
+        if (attempt < 2) {
+          LIBBPF_OPTS(bpf_btf_load_opts,
+                      btf_opts,
+                      .log_buf = log_buf.get(),
+                      .log_level = static_cast<__u32>(log_level),
+                      .log_size = static_cast<__u32>(log_buf_size), );
+
+          auto &btf = prog_.getBTF();
+          btf_fd = bpf_btf_load(btf.data(), btf.size(), &btf_opts);
+          if (btf_fd >= 0)
+            opts.prog_btf_fd = (__u32)btf_fd;
```

Comment:
Could we break the outer (attempt) loop if `btf_fd < 0` here?

Reply:
Yes, I hadn't added the outer loop yet, but I added it now. Done!